### DISCUSSION TITLE
How would I go to pickle a Message?

### DISCUSSION THREAD

**Discussion by DavMrc** [Rapptz/discord.py#7029] [Q&A]
I was curious to know if there was a way to store Messages on the disk so that it is possible to reconstruct it later without manual JSON encoding.
I have tried `pickle` but it raises a `TypeError: cannot pickle '_OverlappedFuture' object`

**Comment by lgaan** [comment]
The library's objects are not serializable. Instead save the useful info (In your case message ID and channel ID); and refetch later with `Bot.get_channel` -> `TextChannel.fetch_message`.

**Comment by adhxtry** [comment]
`pickle_string= pickle.dumps(object)`
Use this pickle string and save it as bytea in postgres, this should work.
To reencode it use:
`solved_pickle_string= pickle.loads(decoded_result)`

**Reply by TheRockettek** [reply]
Please don't use pickle on message objects.

**Reply by Vexs** [reply]
This won't work. Discord.py's objects are stateful, and have references to the underlying, you know, state. They cannot be pickled. 

You have to store information about the object, and then use it to recreate the object.
